test/e2e/managementstate_test.go

```diff
@@ -0,0 +1,141 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/util/retry"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+)
+
+func TestManagementStateUnmanaged(t *testing.T) {
+	client := framework.MustNewClientset(t, nil)
+
+	defer framework.MustRemoveImageRegistry(t, client)
+
+	framework.MustDeployImageRegistry(t, client, &imageregistryv1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imageregistryv1.ImageRegistryResourceName,
+		},
+		Spec: imageregistryv1.ImageRegistrySpec{
+			ManagementState: operatorapi.Managed,
+			Replicas:        1,
+		},
+	})
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+
+	var cr *imageregistryv1.Config
+	var err error
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		cr, err = client.Configs().Get(imageregistryv1.ImageRegistryResourceName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cr.Spec.ManagementState = operatorapi.Unmanaged
+
+		cr, err = client.Configs().Update(cr)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO(dmage): wait for the resource to be observed
```

Comment:
Are we planning to wait for this one before merging? This could generate some flake events, no?

Reply:
No, we have retries below.

Reply:
So we don't need the TODO right? Or I misunderstand you?